### PR TITLE
[issue-57] Clean Component(s), Login, Signup; Setup firestore; Save signup data, fav recipes

### PR DESCRIPTION
**Pull request by looper-m**

### CODE REVIEW

File: src/stories/pages/Home/Home.js [hunk 1]
```diff
@@ -113,7 +135,8 @@ export const Home = () => {
           popularRecipes.map((recipe) => {
             return ({
               id: recipe.id,
-              isFavorite: false,
+              isFavorite: profile &&
+                profile.favoriteRecipes.includes(recipe.id),
```
Comment (alecherryy) [inline src/stories/pages/Home/Home.js:139]: This is throwing an error, need to update to:
```
isFavorite={profile && profile.favoriteRecipes && profile.favoriteRecipes.includes(recipe.id)}
```

File: src/stories/pages/Home/Home.js [hunk 2]
```diff
@@ -140,7 +163,9 @@ export const Home = () => {
             popularRecipes.map((recipe, index) =>
               <Card
                 key={index}
-                isFavorite={false}
+                id={recipe.id}
+                isFavorite={profile &&
+                  profile.favoriteRecipes.includes(recipe.id)}
```
Comment (alecherryy) [inline src/stories/pages/Home/Home.js:168]: Same as above.

File: src/stories/pages/Home/Home.js [hunk 3]
```diff
@@ -113,7 +135,8 @@ export const Home = () => {
           popularRecipes.map((recipe) => {
             return ({
               id: recipe.id,
-              isFavorite: false,
+              isFavorite: profile &&
+                profile.favoriteRecipes.includes(recipe.id),
               url: `/recipes/${recipe.id}`,
               image: `https://spoonacular.com/recipeImages/${recipe.id}-636x393.jpg`,
```
Comment (alecherryy) [inline src/stories/pages/Home/Home.js:141]: You can probably create a global function/variable for the image source. DRYer code.

File: src/stories/pages/Login/Login.js [hunk 4]
```diff
@@ -24,47 +23,30 @@ import { Button } from '../../components/Button/Button';
...
-   */
   const handleLogin = () => {
-    fire
-      .auth()
-      .signInWithEmailAndPassword(email, password)
-      .catch((err) => {
-        switch (err.code) {
-        case 'auth/invalid-email':
-        case 'auth/user-disabled':
-        case 'auth/user-not-found':
```
Comment (alecherryy) [inline src/stories/pages/Login/Login.js:41]: I wonder if at some point we should match this to HTTP error codes.

**Review by alecherryy — CHANGES_REQUESTED** [review]

File: src/stories/pages/Login/Login.js (hunk 4 above)
Comment (looper-m) [reply src/stories/pages/Login/Login.js:41]: Yeah we should. Once we have an error component, we can send a custom message based on returned error code.

**Review by alecherryy — APPROVED** [review]

**Pull request merged**